### ISSUE TITLE
HEVC VA-API encoding not working on RX 590

### ISSUE THREAD

**Issue by tabithamoon** [WiVRn/WiVRn#183]
Attempting to use HEVC hardware encoding via VA-API on a Polaris GPU (in this case, an RX 590) results in no video being sent to the headset.
This GPU requires the use of HEVC due to its slow H.264 encoder, which means WiVRn is practically unusable without huge CPU usage by doing software encoding via x264.
Tested with Flatpak and AUR stable v0.21.1.

Distro: Arch Linux
Headset: Meta Quest 3
Runtime logs: [wivrn.log](https://github.com/user-attachments/files/17850909/wivrn.log)

**Comment by xytovl** [comment]
Logs do not show any issue with the encoder.

If you run with `WIVRN_DUMP_VIDEO=/tmp/wivrn`, you can check the file created at `/tmp/wivrn-0.h264`

**Comment by xytovl** [comment]
To confirm the issue is on the encoder, please try with x264.
Also what is the output of wlx-overlay? You can also try with other test applications such as helo_xr

**Comment by tabithamoon** [comment]
x264 works with no problems, other than the high CPU utilization of course. 
WlxOverlay-S shows no errors, however the log is attached.

[wlxoverlay.txt](https://github.com/user-attachments/files/17852086/wlxoverlay.txt)

I believe this is an encoder bug as when testing with another application (in a separate instance, VRChat thru OpenComposite) the game receives controller inputs and head position data normally, however in the headset there is still no video feed.

**Comment by xytovl** [comment]
I am afraid I won't be able to help more, there is no reported error so I don't even know what to fix.
You can try to set the highest log level for ffmpeg: `FFMPEG_LOG_LEVEL=trace`

**Comment by tabithamoon** [comment]
Here is that log.
[wivrn.txt](https://github.com/user-attachments/files/17852221/wivrn.txt)

Like I mentioned, is WiVRn using 10-bit HEVC? If so that could definitely be the issue, as Polaris does not support this.

**Comment by xytovl** [comment]
It's 8bit only

**Comment by damaestro** [comment]
Please try hardware encoding directly with ffmpeg https://github.com/alvr-org/ALVR/wiki/FFmpeg-Hardware-Encoding-Testing#amd-gpu and report back.

**Comment by tabithamoon** [comment]
HEVC encoding works no problems.

```
Input #0, lavfi, from 'testsrc2=duration=300:size=1280x720:rate=90':
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: wrapped_avframe, yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 90 fps, 90 tbr, 90 tbn
Stream mapping:
  Stream #0:0 -> #0:0 (wrapped_avframe (native) -> hevc (hevc_vaapi))
Press [q] to stop, [?] for help
[hevc_vaapi @ 0x578289e33b80] Driver does not advertise encoder features, using guessed defaults.
[hevc_vaapi @ 0x578289e33b80] Driver does not advertise encoder block size, using guessed defaults.
[hevc_vaapi @ 0x578289e33b80] Driver does not support some wanted packed headers (wanted 0xd, found 0x1).
Output #0, mp4, to 'amd-hevc_vaapi-90fps-300s.mp4':
  Metadata:
    encoder         : Lavf61.7.100
  Stream #0:0: Video: hevc (Main) (hev1 / 0x31766568), vaapi(tv, progressive), 1280x720 [SAR 1:1 DAR 16:9], q=2-31, 90 fps, 11520 tbn
      Metadata:
        encoder         : Lavc61.19.100 hevc_vaapi
[out#0/mp4 @ 0x578289e36040] video:801786KiB audio:0KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 0.014098%
frame=27000 fps=206 q=-0.0 Lsize=  801899KiB time=00:05:00.00 bitrate=21897.2kbits/s speed=2.28x 
```

**Comment by manfredg** [comment]
I have the same problem with a Radeon RX580 GPU. Did anyone ever get to the bottom of this issue?

**Comment by galister** [comment]
Does vulkan encoder also not work with h265?

**Comment by manfredg** [comment]
Nope, Vulkan doesn't work either. I should mention that ALVR does work with VAAPI and HEVC and quite well too. I would have liked to get rid of its reliance on StemVR though.

**Comment by galister** [comment]
Hi, is this still an issue with version 25.9?

**Comment by galister** [comment]
There hasn't been a response for over a month so I'll be closing this.

Vulkan encoder now works with h265 as of current master (not 25.9). If you still have issues, and Vulkan h265 also doesn't work, feel free to create a follow-up.
